### PR TITLE
added logic for stats logging tests

### PR DESCRIPTION
**Pull request by sridevi-15**
#### Fixes #.

- [x] Ran `flake8`
- [ ] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- added logic for stats logging
- Added verification and new test for stats logging

### CODE REVIEW

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_no_conflicts_cbl.py [hunk 1]
```diff
@@ -577,6 +577,18 @@ def test_sg_CBL_updates_concurrently(params_from_base_test_setup, sg_conf_name,
     c = cluster.Cluster(config=cluster_config)
     c.reset(sg_config_path=sg_config)
 
+    if sync_gateway_version >= "2.5.0":
+        sg_client = MobileRestClient()
+        expvars = sg_client.get_expvars(sg_admin_url)
+        num_replications_active = expvars["syncgateway"]["per_db"][sg_db]["database"]["num_replications_active"]
+        num_replications_total = expvars["syncgateway"]["per_db"][sg_db]["database"]["num_replications_total"]
+        """if continuous:
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_no_conflicts_cbl.py]: waiting for the bugs to be fixed. ONce bugs are fixed, need to verify whether having continuous and without continuous replication matters

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py [hunk 2]
```diff
@@ -173,15 +173,17 @@ def test_replication_configuration_with_pull_replication(params_from_base_test_s
...
+                                                                     replication_type="pull", channels=channels, replicator_authenticator_type=authenticator_type,
+                                                                     attachments_generator=attachments_generator)
     sg_docs = sg_client.get_all_docs(url=sg_admin_url, db=sg_db)
     if sg_mode == "di":
         cookie, session = sg_client.create_session(sg_admin_url, sg_db, "autotest")
         authenticator = Authenticator(base_url)
         replicator_authenticator = authenticator.authentication(session, cookie, authentication_type="session")
         replicator = Replication(base_url)
-        replicator.configure_and_replicate(cbl_db, replicator_authenticator, target_url=sg_blip_url, continuous=False,
+        replicator.configure_and_replicate(cbl_db, replicator_authenticator, target_url=sg_blip_url, continuous=True,
```
Comment (hrajput89) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py]: why the change from one-shot to continuous replication?

File: testsuites/syncgateway/functional/tests/test_xattrs.py [hunk 3]
```diff
@@ -2518,3 +2527,123 @@ def update_prop():
...
+@pytest.mark.xattrs
+@pytest.mark.changes
+@pytest.mark.session
+@pytest.mark.parametrize(
+    'sg_conf_name, number_docs_per_client, number_updates_per_doc_per_client',
+    [
+        ('custom_sync/sync_gateway_custom_sync_require_roles', 10, 10),
+    ]
+)
+def test_stats_logging_import_count(params_from_base_test_setup,
```
Comment (hrajput89) [inline testsuites/syncgateway/functional/tests/test_xattrs.py]: if this test is only valid for sg 2.5+, do add skip statement for below versions

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_sg_replicate.py [hunk 4]
```diff
@@ -34,10 +34,11 @@
 @pytest.mark.channel
 @pytest.mark.basicauth
 @pytest.mark.changes
-def test_sg_replicate_basic_test(params_from_base_test_setup):
+def test_sg_replicate_basic_test1(params_from_base_test_setup):
```
Comment (hrajput89) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_sg_replicate.py]: is 1 in test name, typo?

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_sg_replicate.py [hunk 5]
```diff
@@ -338,8 +376,6 @@ def test_sg_replicate_non_existent_db(params_from_base_test_setup):
 ])
 def test_sg_replicate_push_async(params_from_base_test_setup, num_docs):
 
-    assert num_docs > 0
```
Comment (hrajput89) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_sg_replicate.py]: why did we remove assert here?

Comment (sridevi-15) [reply testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_sg_replicate.py]: This is not required assert. Assert at the begining shouldn't be there. We pass the paramemters with num_docs more than 0.  Looks like it was added mistakenly

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py (hunk 2 above)
Comment (sridevi-15) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py]: changed it back to False

File: testsuites/syncgateway/functional/tests/test_xattrs.py (hunk 3 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_xattrs.py]: we can run this test for version below 2.5. It has flow which covers error scenario which is not covered yet in existing SGW test flows

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_sg_replicate.py (hunk 4 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/topology_specific_tests/multiple_sync_gateways/test_sg_replicate.py]: removed 1
